src/components/molecules/ResourceGraphTab/ResourceGraphTab.tsx

```diff
@@ -19,33 +20,46 @@ const ResourceGraphTab: React.FC = () => {
   const dispatch = useAppDispatch();
   const activeStorage = useAppSelector(activeResourceStorageSelector);
   const selectedResource = useSelectedResource();
-  const resources = useMemo(() => (selectedResource ? [selectedResource] : []), [selectedResource]);
   const activeResoureMetaMap = useActiveResourceMetaMap();
   const activeResoureContentMap = useActiveResourceContentMap();
+  const selection = useAppSelector(state => state.main.selection);
```

Comment:
Lets use here `useSelectorWithRef` and use `selectionRef` for `onSelectResource` and `onSelectImage`